cmd/app/add.go

```diff
@@ -72,51 +74,73 @@ func NewAddCommand(clients *shared.ClientFactory) *cobra.Command {
 	}
 
 	cmd.Flags().StringVar(&addFlags.orgGrantWorkspaceID, cmdutil.OrgGrantWorkspaceFlag, "", cmdutil.OrgGrantWorkspaceDescription())
+	cmd.Flags().StringVarP(&addFlags.environmentFlag, "environment", "E", "", "environment of app (local, deployed)")
 
 	return cmd
 }
 
 // preRunAddCommand confirms an app is available for installation
-func preRunAddCommand(ctx context.Context, clients *shared.ClientFactory) error {
+func preRunAddCommand(ctx context.Context, clients *shared.ClientFactory, cmd *cobra.Command) error {
 	err := cmdutil.IsValidProjectDirectory(clients)
 	if err != nil {
 		return err
 	}
 	if !clients.Config.WithExperimentOn(experiment.BoltFrameworks) {
 		return nil
 	}
-	manifestSource, err := clients.Config.ProjectConfig.GetManifestSource(ctx)
-	if err != nil {
-		return err
-	}
-	if manifestSource.Equals(config.ManifestSourceRemote) {
-		return slackerror.New(slackerror.ErrAppInstall).
-			WithMessage("Apps cannot be installed due to project configurations").
-			WithRemediation(
-				"Install an app on app settings: %s\nLink an app to this project with %s\nList apps saved with this project using %s",
-				style.LinkText("https://api.slack.com/apps"),
-				style.Commandf("app link", false),
-				style.Commandf("app list", false),
-			).
-			WithDetails(slackerror.ErrorDetails{
-				slackerror.ErrorDetail{
-					Code:    slackerror.ErrProjectConfigManifestSource,
-					Message: "Cannot install apps with manifests sourced from app settings",
-				},
-			})
-	}
+	clients.Config.SetFlags(cmd)
 	return nil
 }
 
 // RunAddCommand executes the workspace install command, prints output, and returns any errors.
 func RunAddCommand(ctx context.Context, clients *shared.ClientFactory, selection *prompts.SelectedApp, orgGrantWorkspaceID string) (context.Context, types.InstallState, types.App, error) {
 	if selection == nil {
-		selected, err := appSelectPromptFunc(ctx, clients, prompts.ShowHostedOnly, prompts.ShowAllApps)
+		// When team flag is provided, default app environment to deployed if not specified.
+		// TODO(semver:major): Remove defaulting to deployed and require the environment flag to be set.
+		if clients.Config.TeamFlag != "" && addFlags.environmentFlag == "" {
+			err := clients.Config.Flags.Lookup("environment").Value.Set("deployed")
+			if err != nil {
+				return ctx, "", types.App{}, err
+			}
+			clients.Config.Flags.Lookup("environment").Changed = true
+
+			clients.IO.PrintInfo(ctx, false, "\n"+style.Sectionf(style.TextSection{
+				Emoji: "warning",
+				Text:  "Warning: Default App Environment",
+				Secondary: []string{
+					"App environment is set to deployed when only the --team flag is provided.",
+					"The next major version will change this behavior.",
+					"When the --team flag is provided, the --environment flag will be required.",
+					"Add the '--environment deployed' to avoid breaking changes.",
+				},
+			}))
+		}
```

Comment:
note: A warning is now displayed that in the next major release, we'll require the `--environment` flag when the `--team` flag is provided.

<img width="762" height="593" alt="Image" src="https://github.com/user-attachments/assets/eac92fec-e312-48c3-b1b8-81f2acef8b43" />

The double newline haunts me 😱 but it only appear when the warning is shown... which will be rare.